bcipy/acquisition/tests/datastream/test_producer.py

```diff
@@ -64,3 +64,30 @@ def test_max_iters(self):
         self.assertTrue(expected_n - tolerance > maxiters)
         data_n = data_queue.qsize()
         self.assertEqual(data_n, maxiters)
+
+    def test_producer_end(self):
```

Comment:
It is not clear that this test needs to examine multiple different scenarios; perhaps just a generator comprehension `(1 for _ in range(10))` would be enough - but I kept these to show how `StopIteration` seems to NOT be converted into `RuntimeError`